tests/domains_test.go

```diff
@@ -1,34 +1,129 @@
 package tests
 
 import (
+	"fmt"
+	"math/rand"
+	"net/http"
+	"os"
+	"strconv"
+
 	. "github.com/onsi/ginkgo"
-	// . "github.com/onsi/gomega"
+	. "github.com/onsi/gomega"
+	. "github.com/onsi/gomega/gbytes"
+	. "github.com/onsi/gomega/gexec"
 )
 
+func getRandDomain() string {
+	return fmt.Sprintf("my-custom-%d.domain.com", rand.Intn(999999999))
+}
+
 var _ = Describe("Domains", func() {
+	var testApp App
+	var domain string
+
+	Context("with app yet to be deployed", func() {
+
+		BeforeEach(func() {
+			domain = getRandDomain()
+			gitInit()
+
+			testApp.Name = getRandAppName()
+			cmd := createApp(testApp.Name)
+			Eventually(cmd).Should(SatisfyAll(
+				Say("Git remote deis added"),
+				Say("remote available at ")))
+		})
+
+		AfterEach(func() {
+			destroyApp(testApp)
+			gitClean()
+		})
+
+		It("can list domains", func() {
+			sess, err := start("deis domains:list --app=%s", testApp.Name)
+			Expect(err).NotTo(HaveOccurred())
+			Eventually(sess).Should(Say("=== %s Domains", testApp.Name))
+			Eventually(sess).ShouldNot(Say("%s", testApp.Name))
+			Eventually(sess).Should(Exit(0))
+		})
+
+		It("cannot add or remove domains", func() {
```

Comment:
Okay cool - in upcoming PRs (we should merge this one first) I am going to remove the need for that